### PR TITLE
Short name adaptation

### PR DESCRIPTION
**Pull request by pgunnell**
Modification of the short name in a different way (see main.py). 

Do you have any suggestion how to do it differently?

If what is coded now is fine, I can add more short names.

### CODE REVIEW

**Comment by justinasr** [comment]
Remove cookie.txt file from the commit.

File: main.py [hunk 1]
```diff
@@ -85,10 +85,18 @@ def split_chained_request_name(name):
 
 def get_short_name(name):
     spl = name.split('_')
-    renames = {'b': 'bbbar4l',
-               'ST': 'SingleTop'}
-    return renames.get(spl[0], spl[0])
-
+    short_name = spl
+
+    if(spl == 'b'):
```
Comment (justinasr) [inline main.py]: Have you tried if this works? If you split `b_bbar_4l_TuneCP5_13TeV-powheg-pythia8`, you get a list like this:
```
['b', 'bbar', '4l', 'TuneCP5', '13TeV-powheg-pythia8']
```
Which will never be equal to a string `b`.

File: main.py [hunk 2]
```diff
@@ -85,10 +85,18 @@ def split_chained_request_name(name):
...
+        short_name = 'bbbar4l'
+    elif(spl == 'ST'):
+        short_name = 'SingleTop'
+    elif(spl == 'QCD' and 'Flat' in name):
+        short_name = 'QCD P8'
+    elif(spl == 'QCD' and 'madgraph' in name):
+        short_name = 'QCD MG+P8'
+
+    return short_name
```
Comment (justinasr) [inline main.py]: If none of the ifs were true, this will return a list instead of a string.

File: main.py (hunk 1 above)
Comment (justinasr) [reply main.py]: Also, you don't need parentheses in these if statements.

File: update_UL17based.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import sys
+sys.path.append('/afs/cern.ch/cms/PPD/PdmV/tools/McM/')
+from rest import McM
+
+sys.stdout.flush()
+
+mcm = McM(dev=False)
+
+pwgs = ['TOP']
```
Comment (justinasr) [inline update_UL17based.py]: Shouldn't there be more PWGs?

File: MainSamplesFall18.txt [hunk 4]
```diff
@@ -1,277 +1,342 @@
...
-ZJetsToNuNu_HT-200To400_13TeV-madgraph	0	25000000	RunIIFall18wmLHEGS	SUS
-ZZTo2L2Nu_TuneCP5_13TeV_powheg_pythia8	2	50000000	RunIIFall18wmLHEGS	HIG
-ZZTo2L2Q_13TeV_amcatnloFXFX_madspin_pythia8	0	30000000	RunIIFall18wmLHEGS	HIG
-ZZTo2Q2Nu_TuneCP5_13TeV_amcatnloFXFX_madspin_pythia8	0	60000000	RunIIFall18wmLHEGS	SMP
-ZZTo4L_13TeV_powheg_pythia8_TuneCP5	0	20000000	RunIIFall18wmLHEGS	HIG
-ZZTo4L_M-1toInf_TuneCP5_13TeV_powheg_pythia8	0	72500000	RunIIFall18wmLHEGS	SUS
-ZZTo4L_TuneCP5_13TeV-amcatnloFXFX-pythia8	0	27000000	RunIIFall18wmLHEGS	SMP
-ZZTo4L_TuneCP5_13TeV_powheg_pythia8	2	100000000	RunIIFall18wmLHEGS	HIG
\ No newline at end of file
+TT_Mtt-1000toInf_TuneCP5_PSweights_13TeV-powheg-pythia8 0 24000000 RunIIFall18wmLHEGS B2G B2G-RunIIFall18wmLHEGS-00046
```
Comment (justinasr) [inline MainSamplesFall18.txt]: I see that this file is generated using update_twiki.py, so there is no point in storing it in git repository.

File: main.py (hunk 2 above)
Comment (pgunnell) [reply main.py]: ok..thanks! but do you agree with the logic of having many ifs like this?

Comment (justinasr) [reply main.py:99]: Yes, I cannot think of simple, but better solution at the moment. These "short names" are exceptional cases and you have to have all these cases defined (hard-coded) somewhere.

**Comment by justinasr** [comment]
Almost good!
```
************* Module update_UL17based
C:  9, 0: Trailing whitespace (trailing-whitespace)
C: 10, 0: Trailing whitespace (trailing-whitespace)
C: 11, 0: Trailing whitespace (trailing-whitespace)
C: 12, 0: Trailing whitespace (trailing-whitespace)
C: 13, 0: Trailing whitespace (trailing-whitespace)
C: 14, 0: Trailing whitespace (trailing-whitespace)
C: 15, 0: Trailing whitespace (trailing-whitespace)
C: 17, 0: Trailing whitespace (trailing-whitespace)
C: 18, 0: Trailing whitespace (trailing-whitespace)
C: 19, 0: Trailing whitespace (trailing-whitespace)
C: 20, 0: Trailing whitespace (trailing-whitespace)
C: 21, 0: Trailing whitespace (trailing-whitespace)
C: 22, 0: Trailing whitespace (trailing-whitespace)
C: 23, 0: Trailing whitespace (trailing-whitespace)
C: 24, 0: Trailing whitespace (trailing-whitespace)
C: 25, 0: Trailing whitespace (trailing-whitespace)
C: 26, 0: Trailing whitespace (trailing-whitespace)
C: 27, 0: Trailing whitespace (trailing-whitespace)
C: 28, 0: Trailing whitespace (trailing-whitespace)
C: 29, 0: Trailing whitespace (trailing-whitespace)
C: 30, 0: Trailing whitespace (trailing-whitespace)
C: 42, 0: Trailing whitespace (trailing-whitespace)
C: 47, 0: Trailing whitespace (trailing-whitespace)
C: 48, 0: Trailing whitespace (trailing-whitespace)
C: 50, 0: Trailing whitespace (trailing-whitespace)
C: 53, 0: Unnecessary parens after 'if' keyword (superfluous-parens)
C: 53, 0: Exactly one space required around comparison
            if(request_UL18['dataset_name']==dataset_name):
                                           ^^ (bad-whitespace)
C: 55, 0: Trailing whitespace (trailing-whitespace)
C: 60, 0: Trailing whitespace (trailing-whitespace)
C: 61, 0: Unnecessary parens after 'if' keyword (superfluous-parens)
C: 61, 0: Exactly one space required around comparison
            if(request_UL16['dataset_name']==dataset_name):
                                           ^^ (bad-whitespace)
C: 63, 0: Trailing whitespace (trailing-whitespace)
C: 66, 0: Trailing whitespace (trailing-whitespace)
```

**Pull request merged**